### ISSUE TITLE
Input: Various accessibility features

### ISSUE THREAD

**Issue by ArakTaiRoth** [GovAlta/ui-components#2028]
# Info

We got a couple requests for enhanced accessibility from Bruno Alvez. These can be seen [here](https://www.figma.com/design/YgSvDRSpUZK23MzFsYoMxH?node-id=4378-425782#885365350).

## Acceptance Criteria:

1. Required input fields should be using `aria-required=true`
2. Input fields with an error state should be using `aria-invalid=true`
3. Input fields should be using `aria-described-by` linked to the helper and error text
    - Test that the helper text doesn't override the label and both are read out
    - Test that the error text is read out when displayed, or if we need a live region around that
